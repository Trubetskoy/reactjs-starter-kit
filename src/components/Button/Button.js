import React from 'react'

const Button = ({children = 'Submit', callback = () => {}, color = 'blue'}) => {
  return (
    <button data-testid='button' className={`button-${color}`} onClick={callback}>{children}</button>
  )
}

export default Button
