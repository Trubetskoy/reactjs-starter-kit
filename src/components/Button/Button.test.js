/* eslint-env jest */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from './Button'

describe('<Button /> tests', () => {
  it('should render as default button', () => {
    const color = 'red'
    const children = 'test'

    const { container } = render(<Button color={`${color}`} callback={() => {}}>{children}</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('button should has class', () => {
    const color = 'red'

    const { container } = render(<Button color={`${color}`} callback={() => {}} />)
    expect(container.firstChild).toHaveClass(`button-${color}`)
  })

  it('button should has children', () => {
    const color = 'red'
    const children = 'test'

    const { getByText } = render(<Button color={`${color}`} callback={() => {}}>{children}</Button>)
    getByText(children)
  })

  it('button should call callback on click', () => {
    const color = 'red'
    const children = 'test'
    const content = 'button'
    const onClick = jest.fn()

    const { getByTestId } = render(<Button color={`${color}`} callback={onClick}>{children}</Button>)
    fireEvent.click(getByTestId(content))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
