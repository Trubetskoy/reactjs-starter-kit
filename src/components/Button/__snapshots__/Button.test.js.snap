// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should render as default button 1`] = `
<button
  class="button-red"
  data-testid="button"
>
  test
</button>
`;

exports[`<Button /> tests should render as default button 1 1`] = `
<button
  class="button-red"
  data-testid="button"
>
  test
</button>
`;
